pkg/daemon/cleanup/cleanup.go

```diff
@@ -52,6 +55,23 @@ func Deconfigure(cfg *rest.Config, ctx context.Context, configMapNamespace strin
 	if err != nil {
 		klog.Infof("no associated node found")
 	}
+	if node != nil {
+		k8sClient, err := kubernetes.NewForConfig(cfg)
+		if err != nil {
+			return fmt.Errorf("unable to initialize client for drainer")
+		}
+		drainer := drainhelper.NewDrainHelper(ctx, k8sClient, klog.NewKlogr())
+		// Make a best effort attempt to cordon and drain the node before removing any services. We shouldn't block
+		// node cleanup if this fails.
+		err = drain.RunCordonOrUncordon(drainer, node, true)
+		if err != nil {
+			klog.Errorf("unable to cordon node: %s", err)
+		}
+		err = drain.RunNodeDrain(drainer, node.GetName())
+		if err != nil {
+			klog.Errorf("unable to drain node: %s", err)
+		}
+	}
```

Comment:
Please make this a helper method. We should be able to glance at `Deconfigure()` and understand what it does.